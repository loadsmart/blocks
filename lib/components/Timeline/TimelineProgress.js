"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var react_native_1 = require("react-native");
var res_1 = require("../../res");
var lineWidth = 4;
var topPartHeight = 16;
var markerDiameter = 16;
var LineStyle;
(function (LineStyle) {
    LineStyle[LineStyle["Empty"] = 0] = "Empty";
    LineStyle[LineStyle["Filled"] = 1] = "Filled";
})(LineStyle = exports.LineStyle || (exports.LineStyle = {}));
var TimelineProgress = /** @class */ (function (_super) {
    __extends(TimelineProgress, _super);
    function TimelineProgress() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    TimelineProgress.prototype.render = function () {
        return (react_1.default.createElement(react_native_1.View, { style: [styles.container, this.props.style] },
            react_1.default.createElement(react_native_1.View, { accessibilityLabel: 'progress top line', style: [
                    styles.topPart,
                    {
                        backgroundColor: this.lineStyleToColor(this.props.topPartStyle),
                        opacity: this.props.topPartVisible ? 1 : 0,
                    },
                ] }),
            react_1.default.createElement(react_native_1.View, { style: [
                    styles.marker,
                    { backgroundColor: this.lineStyleToColor(this.props.markerStyle) },
                ] }),
            react_1.default.createElement(react_native_1.View, { style: [
                    styles.bottomPart,
                    {
                        backgroundColor: this.lineStyleToColor(this.props.bottomPartStyle),
                        opacity: this.props.bottomPartVisible ? 1 : 0,
                    },
                ] })));
    };
    TimelineProgress.prototype.lineStyleToColor = function (style) {
        return style === LineStyle.Empty ? res_1.Colors.LightGray : res_1.Colors.AlgaeGreen;
    };
    TimelineProgress.defaultProps = {
        topPartStyle: LineStyle.Filled,
        topPartVisible: true,
        markerStyle: LineStyle.Filled,
        bottomPartStyle: LineStyle.Filled,
        bottomPartVisible: true,
    };
    return TimelineProgress;
}(react_1.PureComponent));
exports.TimelineProgress = TimelineProgress;
var styles = react_native_1.StyleSheet.create({
    container: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        minHeight: 2 * topPartHeight + markerDiameter,
    },
    topPart: {
        height: 16,
        width: lineWidth,
        zIndex: 3,
    },
    marker: {
        height: markerDiameter,
        width: markerDiameter,
        borderRadius: markerDiameter / 2.0,
        zIndex: 2,
    },
    bottomPart: {
        flex: 2,
        width: lineWidth,
        zIndex: 1,
    },
});
