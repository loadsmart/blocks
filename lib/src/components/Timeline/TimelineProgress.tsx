import React, { PureComponent } from 'react'
import { StyleSheet, View, ViewProps } from 'react-native'
import { Colors } from '../../res'

const lineWidth = 4
const topPartHeight = 16
const markerDiameter = 16

export enum LineStyle {
  Empty,
  Filled,
}

interface Props extends ViewProps {
  topPartStyle?: LineStyle
  topPartVisible?: boolean
  markerStyle?: LineStyle
  bottomPartStyle?: LineStyle
  bottomPartVisible?: boolean
}

export default class TimelineProgress extends PureComponent<Props> {
  static defaultProps = {
    topPartStyle: LineStyle.Filled,
    topPartVisible: true,
    markerStyle: LineStyle.Filled,
    bottomPartStyle: LineStyle.Filled,
    bottomPartVisible: true,
  }

  public render() {
    return (
      <View style={[styles.container, this.props.style]}>
        <View
          accessibilityLabel={'progress top line'}
          style={[
            styles.topPart,
            {
              backgroundColor: this.lineStyleToColor(this.props.topPartStyle),
              opacity: this.props.topPartVisible ? 1 : 0,
            },
          ]}
        />
        <View
          style={[
            styles.marker,
            { backgroundColor: this.lineStyleToColor(this.props.markerStyle) },
          ]}
        />
        <View
          style={[
            styles.bottomPart,
            {
              backgroundColor: this.lineStyleToColor(this.props.bottomPartStyle),
              opacity: this.props.bottomPartVisible ? 1 : 0,
            },
          ]}
        />
      </View>
    )
  }

  private lineStyleToColor(style?: LineStyle) {
    return style === LineStyle.Empty ? Colors.LightGray : Colors.AlgaeGreen
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    minHeight: 2 * topPartHeight + markerDiameter,
  },
  topPart: {
    height: 16,
    width: lineWidth,
    zIndex: 3,
  },
  marker: {
    height: markerDiameter,
    width: markerDiameter,
    borderRadius: markerDiameter / 2.0,
    zIndex: 2,
  },
  bottomPart: {
    flex: 2,
    width: lineWidth,
    zIndex: 1,
  },
})
