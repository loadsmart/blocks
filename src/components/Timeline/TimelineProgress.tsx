import React, { PureComponent } from 'react'
import { StyleSheet, View, ViewProps } from 'react-native'
import { Colors } from '../../res'
import { Theme, ThemeContext } from '../Contexts'

const lineWidth = 4
const topPartHeight = 16
const markerDiameter = 16

export enum LineStyle {
  Empty,
  Filled,
}

interface Props extends ViewProps {
  topPartStyle?: LineStyle
  topPartVisible?: boolean
  markerStyle?: LineStyle
  bottomPartStyle?: LineStyle
  bottomPartVisible?: boolean
}

export class TimelineProgress extends PureComponent<Props> {
  public static defaultProps = {
    topPartStyle: LineStyle.Filled,
    topPartVisible: true,
    markerStyle: LineStyle.Filled,
    bottomPartStyle: LineStyle.Filled,
    bottomPartVisible: true,
  }

  public render() {
    return (
      <ThemeContext.Consumer>
        {theme => {
          return (
            <View style={[styles.container, this.props.style]}>
              <View
                accessibilityLabel={'progress top line'}
                style={[
                  styles.topPart,
                  {
                    backgroundColor: this.lineStyleToColor(theme, this.props.topPartStyle),
                    opacity: this.props.topPartVisible ? 1 : 0,
                  },
                ]}
              />
              <View
                style={[
                  styles.marker,
                  { backgroundColor: this.lineStyleToColor(theme, this.props.markerStyle) },
                ]}
              />
              <View
                style={[
                  styles.bottomPart,
                  {
                    backgroundColor: this.lineStyleToColor(theme, this.props.bottomPartStyle),
                    opacity: this.props.bottomPartVisible ? 1 : 0,
                  },
                ]}
              />
            </View>
          )
        }}
      </ThemeContext.Consumer>
    )
  }

  private lineStyleToColor(theme: Partial<Theme>, style?: LineStyle) {
    return style === LineStyle.Empty ? Colors.LightGray : theme.primaryColor
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    minHeight: 2 * topPartHeight + markerDiameter,
  },
  topPart: {
    height: 16,
    width: lineWidth,
    zIndex: 3,
  },
  marker: {
    height: markerDiameter,
    width: markerDiameter,
    borderRadius: markerDiameter / 2.0,
    zIndex: 2,
  },
  bottomPart: {
    flex: 2,
    width: lineWidth,
    zIndex: 1,
  },
})
