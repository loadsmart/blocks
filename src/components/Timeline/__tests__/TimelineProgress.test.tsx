import React from 'react'
import renderer from 'react-test-renderer'
import { LineStyle, TimelineProgress } from '../TimelineProgress'

//TODO better to test this check styles?
describe('TimelineProgress', () => {
  describe('topPart', () => {
    it('can be visible', () => {
      const testRenderer = renderer.create(<TimelineProgress topPartVisible={true} />)
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('can be invisible', () => {
      const testRenderer = renderer.create(<TimelineProgress topPartVisible={false} />)
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('can be filled', () => {
      const testRenderer = renderer.create(
        <TimelineProgress topPartVisible={true} topPartStyle={LineStyle.Filled} />
      )
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('can be empty', () => {
      const testRenderer = renderer.create(
        <TimelineProgress topPartVisible={true} topPartStyle={LineStyle.Empty} />
      )
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })

  describe('marker', () => {
    it('can be filled', () => {
      const testRenderer = renderer.create(<TimelineProgress markerStyle={LineStyle.Filled} />)
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('can be empty', () => {
      const testRenderer = renderer.create(<TimelineProgress markerStyle={LineStyle.Empty} />)
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })

  describe('bottomPart', () => {
    it('can be visible', () => {
      const testRenderer = renderer.create(<TimelineProgress bottomPartVisible={true} />)
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('can be invisible', () => {
      const testRenderer = renderer.create(<TimelineProgress bottomPartVisible={false} />)
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('can be filled', () => {
      const testRenderer = renderer.create(
        <TimelineProgress bottomPartVisible={true} bottomPartStyle={LineStyle.Filled} />
      )
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('can be empty', () => {
      const testRenderer = renderer.create(
        <TimelineProgress bottomPartVisible={true} bottomPartStyle={LineStyle.Empty} />
      )
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })
})
