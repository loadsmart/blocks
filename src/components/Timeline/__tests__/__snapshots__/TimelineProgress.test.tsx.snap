// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineProgress bottomPart can be empty 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": 48,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="progress top line"
    style={
      Array [
        Object {
          "height": 16,
          "width": 4,
          "zIndex": 3,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": 16,
          "width": 16,
          "zIndex": 2,
        },
        Object {
          "backgroundColor": "#14d64d",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 2,
          "width": 4,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "#d4d4d4",
          "opacity": 1,
        },
      ]
    }
  />
</View>
`;

exports[`TimelineProgress bottomPart can be filled 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": 48,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="progress top line"
    style={
      Array [
        Object {
          "height": 16,
          "width": 4,
          "zIndex": 3,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": 16,
          "width": 16,
          "zIndex": 2,
        },
        Object {
          "backgroundColor": "#14d64d",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 2,
          "width": 4,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
</View>
`;

exports[`TimelineProgress bottomPart can be invisible 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": 48,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="progress top line"
    style={
      Array [
        Object {
          "height": 16,
          "width": 4,
          "zIndex": 3,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": 16,
          "width": 16,
          "zIndex": 2,
        },
        Object {
          "backgroundColor": "#14d64d",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 2,
          "width": 4,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 0,
        },
      ]
    }
  />
</View>
`;

exports[`TimelineProgress bottomPart can be visible 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": 48,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="progress top line"
    style={
      Array [
        Object {
          "height": 16,
          "width": 4,
          "zIndex": 3,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": 16,
          "width": 16,
          "zIndex": 2,
        },
        Object {
          "backgroundColor": "#14d64d",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 2,
          "width": 4,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
</View>
`;

exports[`TimelineProgress marker can be empty 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": 48,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="progress top line"
    style={
      Array [
        Object {
          "height": 16,
          "width": 4,
          "zIndex": 3,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": 16,
          "width": 16,
          "zIndex": 2,
        },
        Object {
          "backgroundColor": "#d4d4d4",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 2,
          "width": 4,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
</View>
`;

exports[`TimelineProgress marker can be filled 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": 48,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="progress top line"
    style={
      Array [
        Object {
          "height": 16,
          "width": 4,
          "zIndex": 3,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": 16,
          "width": 16,
          "zIndex": 2,
        },
        Object {
          "backgroundColor": "#14d64d",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 2,
          "width": 4,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
</View>
`;

exports[`TimelineProgress topPart can be empty 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": 48,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="progress top line"
    style={
      Array [
        Object {
          "height": 16,
          "width": 4,
          "zIndex": 3,
        },
        Object {
          "backgroundColor": "#d4d4d4",
          "opacity": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": 16,
          "width": 16,
          "zIndex": 2,
        },
        Object {
          "backgroundColor": "#14d64d",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 2,
          "width": 4,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
</View>
`;

exports[`TimelineProgress topPart can be filled 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": 48,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="progress top line"
    style={
      Array [
        Object {
          "height": 16,
          "width": 4,
          "zIndex": 3,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": 16,
          "width": 16,
          "zIndex": 2,
        },
        Object {
          "backgroundColor": "#14d64d",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 2,
          "width": 4,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
</View>
`;

exports[`TimelineProgress topPart can be invisible 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": 48,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="progress top line"
    style={
      Array [
        Object {
          "height": 16,
          "width": 4,
          "zIndex": 3,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 0,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": 16,
          "width": 16,
          "zIndex": 2,
        },
        Object {
          "backgroundColor": "#14d64d",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 2,
          "width": 4,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
</View>
`;

exports[`TimelineProgress topPart can be visible 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": 48,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityLabel="progress top line"
    style={
      Array [
        Object {
          "height": 16,
          "width": 4,
          "zIndex": 3,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": 16,
          "width": 16,
          "zIndex": 2,
        },
        Object {
          "backgroundColor": "#14d64d",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 2,
          "width": 4,
          "zIndex": 1,
        },
        Object {
          "backgroundColor": "#14d64d",
          "opacity": 1,
        },
      ]
    }
  />
</View>
`;
